config/config.go

```diff
@@ -306,15 +314,19 @@ func (cfg *Manager) LoadJSON(bs []byte) error {
 
 	sections := cfg.sections
 	// will skip checking errors and trust Validate()
-	loadCompJSON(sections[Consensus], jcfg.Consensus)
-	loadCompJSON(sections[API], jcfg.API)
-	loadCompJSON(sections[IPFSConn], jcfg.IPFSConn)
-	loadCompJSON(sections[State], jcfg.State)
-	loadCompJSON(sections[PinTracker], jcfg.PinTracker)
-	loadCompJSON(sections[Monitor], jcfg.Monitor)
-	loadCompJSON(sections[Allocator], jcfg.Allocator)
-	loadCompJSON(sections[Informer], jcfg.Informer)
-	loadCompJSON(sections[Sharder], jcfg.Informer)
+	loadSectionJSON(sections[Consensus], jcfg.Consensus)
+	loadSectionJSON(sections[API], jcfg.API)
+	// Should we change hardcoded "ipfsproxy" to something else
+	if _, ok := jcfg.API["ipfsproxy"]; !ok {
+		loadCompJSON("ipfsproxy", sections[API]["ipfsproxy"], jcfg.IPFSConn)
+	}
```

Comment:
This seems a bit hardcoded. Seeing how I can change this.

Reply:
Also, I have utilized the property that superset JSON object can be unmarshalled into subject subset JSON structs

Here, previous IPFSConnector config is superset of new IPFSConnector config and IPFSProxy config. So, if our service.json has old IPFSConnector config, we can unmarshal configs for new IPFSConnector and IPFSProxy.
I don't know how well known this propery is so [here](https://play.golang.org/p/qdO4AJ-QbfG) is an illustration.